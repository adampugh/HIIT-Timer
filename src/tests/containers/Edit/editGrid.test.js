import React from "react";
import { shallow } from "enzyme"; 

import EditGrid from "../../../containers/Edit/editGrid";
import workoutState from "../../fixtures/workouts";

test("should render editGrid correctly", () => {
    const wrapper = shallow(<EditGrid 
        workout={workoutState.workouts[0].exercises} 
        workoutId={workoutState.workouts[0].id} />);
    expect(wrapper).toMatchSnapshot();
});

test("should render editGrid correctly without exercises", () => {
    const wrapper = shallow(<EditGrid 
        workout={[]} 
        workoutId={workoutState.workouts[0].id} />);
    expect(wrapper).toMatchSnapshot();
});

test("should call handleDeleteModal and update state in editGrid", () => {
    const exercises = workoutState.workouts[0].exercises
    const wrapper = shallow(<EditGrid 
        workout={exercises} 
        workoutId={workoutState.workouts[0].id}
        handleOpenDeleteModal={() => {}} />)
    wrapper.find('h3').first().simulate("click");
    expect(wrapper.state('selectedExerciseIndex')).toBe(0);
    expect(wrapper.state('selectedExerciseId')).toBe(exercises[0].id);
});