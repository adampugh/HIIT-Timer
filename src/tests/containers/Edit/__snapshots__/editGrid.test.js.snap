// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render editGrid correctly 1`] = `
<div>
  <Connect(C)
    workoutId="L6ERTEQ"
  />
  <Connect(deleteExerciseModal)
    exerciseId={null}
    exerciseIndex={null}
    modalTitle="Exercise"
    workoutId="L6ERTEQ"
  />
  <div
    className="edit__wrapper"
  >
    <div
      className="container"
    >
      <div
        className="edit__grid"
      >
        <div
          key="squats0"
        >
          <editBlock
            color="#40b3e1"
            name="squats"
            time={60}
          />
          <div
            className="editBlock__delete"
          >
            <h3
              onClick={[Function]}
            >
              <i
                className="far fa-trash-alt"
              />
               Delete
            </h3>
          </div>
        </div>
        <div
          key="pressups1"
        >
          <editBlock
            color="#40b3e1"
            name="pressups"
            time={60}
          />
          <div
            className="editBlock__delete"
          >
            <h3
              onClick={[Function]}
            >
              <i
                className="far fa-trash-alt"
              />
               Delete
            </h3>
          </div>
        </div>
        <div
          className="edit__addBlock"
        >
          <h1>
            Add an Exercise
          </h1>
          <i
            className="fas fa-plus-circle"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render editGrid correctly without exercises 1`] = `
<div>
  <Connect(C)
    workoutId="L6ERTEQ"
  />
  <Connect(deleteExerciseModal)
    exerciseId={null}
    exerciseIndex={null}
    modalTitle="Exercise"
    workoutId="L6ERTEQ"
  />
  <div
    className="edit__wrapper"
  >
    <div
      className="container"
    >
      <div
        className="edit__grid"
      >
        <div
          className="edit__addBlock"
        >
          <h1>
            Add an Exercise
          </h1>
          <i
            className="fas fa-plus-circle"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
